File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.js
function solution(n, arr1, arr2) {
    let answer = []
    for (let i = 0; i < n; i++) {
        let key1 = arr1[i].toString(2).padStart(n, 0)
        let key2 = arr2[i].toString(2).padStart(n, 0)
        let password = ''
        for (let j = 0; j < n; j++) {
            if (key1[j] === '1' || key2[j] === '1') password += '#'
            else password += ' '
        }
        answer.push(password)
    }
    return answer
}

File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.py
def solution(n, arr1, arr2):
    answer = []
    for i in range(n):
        key1 = format(arr1[i], 'b').zfill(n)
        key2 = format(arr2[i], 'b').zfill(n)
        password_row = ''
        for j in range(n):
            if key1[j] == '1' or key2[j] == '1':
                password_row += '#'
            else:
                password_row += ' '
        answer.append(password_row)
    return answer